### PR TITLE
[WIP] Send Feedback To User

### PR DESCRIPTION
**Pull request by CalumTowers**

### CODE REVIEW

**Comment by CalumTowers** [comment]
To do

- Review permissions
- Test

**Comment by CalumTowers** [comment]
Add permission to migration

**Comment by CalumTowers** [comment]
@atoff Have you had a chance to look at this?

File: app/Http/Controllers/Adm/Mship/Feedback/FeedbackSendController.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+use App\Models\Mship\Feedback\Feedback as FeedbackModel;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Redirect;
+
+class FeedbackSendController extends \App\Http\Controllers\BaseController
+{
+    public function store(FeedbackModel $feedback, Request $request)
+    {
+        $feedback->markSent(\Auth::user(), $request->input('comment'));
```
Comment (atoff) [inline app/Http/Controllers/Adm/Mship/Feedback/FeedbackSendController.php]: Might as well use the `use Illuminate\Support\Facades\Auth;` here

File: app/Http/Controllers/Mship/Feedback/ViewFeedbackController.php [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+class ViewFeedbackController extends \App\Http\Controllers\BaseController
+{
+    public function show()
+    {
+        $feedback = $this->account->feedback()
+            ->sent()
+            ->get()
+            ->reverse();
+
+        if ($feedback->count() === 0) {
```
Comment (atoff) [inline app/Http/Controllers/Mship/Feedback/ViewFeedbackController.php]: Could use `$feedback->isEmpty()` here

File: app/Models/Mship/Feedback/Feedback.php [hunk 3]
```diff
@@ -82,6 +86,11 @@ public function scopeUnActioned($query)
         return $query->where('actioned_at', null);
     }
 
+    public function scopeSent($query)
+    {
+        return $query->where('sent_at', '!=', null);
```
Comment (atoff) [inline app/Models/Mship/Feedback/Feedback.php]: Could shorten this to `whereNotNull('sent_at')`. Could refactor that above as well if you find yourself doing nothing...

File: resources/views/adm/mship/feedback/view.blade.php [hunk 4]
```diff
@@ -2,132 +2,179 @@
...
+
+                    <div class="box-body">
+                        <div class="row">
+                            <div class="col-md-12">
+                                @if (!$feedback->sent_at)
+                                {{ Form::open(['route' => ['adm.mship.feedback.send', $feedback->id]]) }}
+                                <div class="col-md-12 text-center">
+                                    <p>
+                                        {{ Form::label('Comment To User')}}
+                                        {{ Form::textarea('comment', null, ['rows' => 4, "cols"=> 10, "class" => "form-control"]) }}
```
Comment (atoff) [inline resources/views/adm/mship/feedback/view.blade.php]: Possibly add a "helper" text here, saying that if left blank, it will advise the member that positive feedback was left about them.

File: resources/views/mship/feedback/view.blade.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
+@extends('layout')
+
+@section('content')
+<div class="panel panel-ukblue">
+	<div class="panel-heading"> My Feedback</div>
+	<div class="panel-body">
+		<p>Here, you can view feedback that has been sent to you by the VATSIM UK training departments.</p>
+        @foreach($feedback as $item)
```
Comment (atoff) [inline resources/views/mship/feedback/view.blade.php]: Personally, I would opt for a table layout here. I feel that it would be easier to read and browse through. I suggest something like this:
image

File: app/Http/Controllers/Adm/Mship/Feedback/FeedbackSendController.php (hunk 1 above)
Comment (CalumTowers) [reply app/Http/Controllers/Adm/Mship/Feedback/FeedbackSendController.php]: f5c3990

File: app/Http/Controllers/Mship/Feedback/ViewFeedbackController.php (hunk 2 above)
Comment (CalumTowers) [reply app/Http/Controllers/Mship/Feedback/ViewFeedbackController.php]: 6502be0

File: app/Models/Mship/Feedback/Feedback.php (hunk 3 above)
Comment (CalumTowers) [reply app/Models/Mship/Feedback/Feedback.php]: 4ecfad7

File: resources/views/adm/mship/feedback/view.blade.php (hunk 4 above)
Comment (CalumTowers) [reply resources/views/adm/mship/feedback/view.blade.php]: 99a20e3

File: resources/views/mship/feedback/view.blade.php (hunk 5 above)
Comment (CalumTowers) [reply resources/views/mship/feedback/view.blade.php]: a24db50
